src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -431,7 +438,15 @@ class GroupReadsByUmi
 
   private val assigner = strategy.newStrategy(this.edits)
 
-  type ReadPair = (SamRecord, SamRecord)
+  /** Checks that the read's mapq is over a minimum, and if the read is paired, that the mate mapq is also over the min. */
+  private def mapqOk(rec: SamRecord, minMapQ: Int): Boolean = {
+    val mateMqOk = if (rec.unpaired) true else {
+      val mateMq = rec.get[Int](SAMTag.MQ.name())
+      if (mateMq.isEmpty) fail(s"") else mateMq.forall(_ >= minMapQ)
```

Comment:
better fail message since no stack trace is given.  Include the read name.   Why not a match statement?
``` 
rec.get[Int](SAMTag.MQ.name()) match {
    case None      => fail(s"Read did not have the mate mapping quality (MQ tag) set: ${rec.name}")
   case Some(mapq) => mapq >= minMapQ
}
```

Reply:
Done.